src/kademlia/dht_storage.cpp

```diff
@@ -235,26 +232,20 @@ namespace
 					if (iter->addr.protocol() != protocol)
```

Comment:
there are these conditions here of peers that are excluded. afaict, these filters will skew the picking and make us potentially pick too few.